### PR TITLE
feat: integrate label parsing to kili.labels()

### PR DESCRIPTION
**Pull request by Jonas1312**
This PR integrates the label parsing feature in the kili.labels() method

Also adds a tutorial for the label parsing

I'm open to suggestions when it comes to the method and parameter names

also add tests for poseestimation labels

### CODE REVIEW

**Review by theodu — CHANGES_REQUESTED** [review]

**Review by PierreLeveau — CHANGES_REQUESTED** [review]

File: docs/sdk/tutorials/label_parsing.md [hunk 1]
```diff
@@ -0,0 +1,1712 @@
...
+
+## .jobs attribute
+
+The `.jobs` attribute of a `ParsedLabel` class is a dictionary-like object that contains the parsed labels.
+
+The keys are the names of the jobs, and the values are the parsed job responses.
+
+Let's create a simple Kili project to illustrate this.
+
+## Classification jobs
```
[the remark this answers is not in the document]
Comment (Jonas1312) [reply docs/sdk/tutorials/label_parsing.md]: Made some modifs, I let you check if it's better

File: docs/sdk/label_utils.md [hunk 2]
```diff
@@ -1,6 +1,14 @@
...
+
+## ParsedLabel
+
+::: kili.utils.labels.parsing.ParsedLabel
+    options:
+        members:
+            - to_dict
+            - "__init__"
 
 ## Points
```
[the remark this answers is not in the document]
Comment (Jonas1312) [reply docs/sdk/label_utils.md:5]: done

File: src/kili/utils/labels/parsing.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+
+class ParsedLabel(Dict):
+    """Class that represents a parsed label."""
+
+    def __init__(self, label: Dict, json_interface: Dict, input_type: InputType) -> None:
+        # pylint: disable=line-too-long
+        """Class that represents a parsed label.
+
+        The class behaves like a dict but adds the attribute `.jobs`.
```
[the remark this answers is not in the document]
Comment (Jonas1312) [reply src/kili/utils/labels/parsing.py:17]: well a ParsedLabel should provide all features available in a "normal" dict label, plus the additional `.jobs` attribute

as discussed on the slack, we could build an ORM over possible keys available in a label dict, but that would require a lot of work for low to zero added value

**Comment by Jonas1312** [comment]
@PierreLeveau how would you want to add the support for parsed label in other kili methods?

I'm not sure if it's a good idea, since other kili methods take json response arrays, not "labels"

so we would need to add another argument "label_array" for each label mutation method

I think it's reasonable to let the user do `label["jsonResponse"]`, which is already the current state

However, we can add a `.json_response` property to the ParsedLabel class, in order to avoid doing `parsed_label.to_dict()["jsonResponse"]` which is quite ugly. We would have instead `parsed_label.json_response`

wdyt @kili-technology/ml ?

**Comment by theodu** [comment]
> @PierreLeveau how would you want to add the support for parsed label in other kili methods?
> 
> I'm not sure if it's a good idea, since other kili methods take json response arrays, not "labels"
> 
> so we would need to add another argument "label_array" for each label mutation method
> 
> I think it's reasonable to let the user do `label["jsonResponse"]`, which is already the current state
> 
> However, we can add a `.json_response` property to the ParsedLabel class, in order to avoid doing `parsed_label.to_dict()["jsonResponse"]` which is quite ugly. We would have instead `parsed_label.json_response`
> 
> wdyt @kili-technology/ml ?

I agree with you for all your remarks

File: src/kili/utils/labels/parsing.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (PierreLeveau) [reply src/kili/utils/labels/parsing.py:17]: 👌

File: recipes/label_parsing.ipynb [hunk 4]
```diff
@@ -0,0 +1,2944 @@
+{
```
Comment (PierreLeveau) [inline recipes/label_parsing.ipynb:1]: <p>This should be the main entrypoint to the parsed label class, so this section should move up, before detailing what a parsed label is and how to use it. Converting from an existing dict should come here. WDYT?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/kili-technology/kili-python-sdk/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

**Review by theodu — APPROVED** [review]

File: recipes/label_parsing.ipynb (hunk 4 above)
Comment (Jonas1312) [reply recipes/label_parsing.ipynb:1]: done

File: recipes/label_parsing.ipynb [hunk 5]
```diff
@@ -0,0 +1,3063 @@
...
```
Comment (xczuba) [inline recipes/label_parsing.ipynb:1]: <p>Will you be adding this tutorial to the TOC listing tutorials in SDK docs? If so, maybe we should also add something to the intro topic that I sent you today?</p><p>Also: how about adding a summary list of items covered in this NB? Other NBs see to have one.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/kili-technology/kili-python-sdk/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (xczuba) [inline recipes/label_parsing.ipynb:1]: <p>"that allows to automatically parse the labels" -&gt; "that enables automatic label parsing"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/kili-technology/kili-python-sdk/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (xczuba) [inline recipes/label_parsing.ipynb:1]: <p>"Task specific" -&gt; "Task-specific"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/kili-technology/kili-python-sdk/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (xczuba) [inline recipes/label_parsing.ipynb:1]: <p>"for a two classification jobs:" -&gt; "for two classification jobs"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/kili-technology/kili-python-sdk/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

Comment (xczuba) [inline recipes/label_parsing.ipynb:1]: <p>"for multi-categories classification jobs" -&gt; "for multi-category classification jobs"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/kili-technology/kili-python-sdk/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='47'/>

Comment (xczuba) [inline recipes/label_parsing.ipynb:1]: <p>Suggestion:</p><p> </p><p>"the <code>.text</code> allows to access the label data" -&gt; "you can access the label data through the .text attribute"</p><p> </p><p>WDYT?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/kili-technology/kili-python-sdk/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='52'/>

Comment (xczuba) [inline recipes/label_parsing.ipynb:1]: <p>"The point coordinates of a point detection label is accessible" -&gt; "The point coordinates of a point detection label are accessible"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/kili-technology/kili-python-sdk/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='68'/>

Comment (xczuba) [inline recipes/label_parsing.ipynb:1]: <p>Suggestion:</p><p> </p><p>"that returns a list of the points composing the annotated object" -&gt; "that returns the list of points that make up the annotation"</p><p> </p><p>WDYT?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/kili-technology/kili-python-sdk/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='82'/>

Comment (xczuba) [inline recipes/label_parsing.ipynb:1]: <p>When do you use entity_annotations? There's no example showing how to use this one.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/kili-technology/kili-python-sdk/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='101'/>

Comment (xczuba) [inline recipes/label_parsing.ipynb:1]: <p>"A NER in PDFs parsed label " -&gt; "For PDF assets, a parsed NER label "</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/kili-technology/kili-python-sdk/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='112'/>

Comment (xczuba) [inline recipes/label_parsing.ipynb:1]: <p>"The NER in PDFs json response format is a bit complex, and thus requires to use the <code>.annotations</code> attribute a second time." -&gt; "When exported as json_response, the format of NER annotations added to PDF assets is a bit complex and requires using the <code>.annotations</code> attribute for the second time."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/kili-technology/kili-python-sdk/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='119'/>

Comment (xczuba) [inline recipes/label_parsing.ipynb:1]: <p>"Below, we print the entity recognition annotations unique IDs:" -&gt; "Here's how to print the unique IDs of the entity recognition annotations""</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/kili-technology/kili-python-sdk/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='130'/>

Comment (xczuba) [inline recipes/label_parsing.ipynb:1]: <p>"Children jobs" -&gt; "Child jobs"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/kili-technology/kili-python-sdk/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='141'/>

Comment (xczuba) [inline recipes/label_parsing.ipynb:1]: <p>"Children jobs" -&gt; "Child jobs"</p><p> </p><p>"For children jobs, the <code>.children</code> attribute allows to access the children labels" -&gt; "Use the <code>.children</code> attribute to access nested (child) labels"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/kili-technology/kili-python-sdk/pull/1198/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='142'/>

**Pull request merged**